dot/network/README.md

```diff
@@ -0,0 +1,178 @@
+network use to securely exchange public keys - `libp2p` supports [Noise](#noise), a key distribution framework that is
+based on [Diffie-Hellman key exchange](https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange).
+
+### Peer Discovery & Management
+
+In a peer-to-peer network, "[discovery](https://docs.libp2p.io/concepts/publish-subscribe/#discovery)" is the term that
+is used to describe the mechanism that peers use to find one another - this is an important topic since there is not a
+privileged authority that can maintain an index of known/trusted network participants. The discovery mechanisms that
+peer-to-peer networks use have evolved over time - [Napster](https://en.wikipedia.org/wiki/Napster) relied on a central
+database, [Gnutella](https://en.wikipedia.org/wiki/Gnutella) used a brute-force technique called "flooding",
+[BitTorrent](https://en.wikipedia.org/wiki/BitTorrent) takes a performance-preserving approach that relies on a
+[distributed hash table (DHT)](https://en.wikipedia.org/wiki/Distributed_hash_table). Gossamer uses a `libp2p`-based
+implementation of the [Kademlia](#kademlia) DHT for peer discovery.
+
+### Stream Multiplexing
+
+[Multiplexing](https://en.wikipedia.org/wiki/Multiplexing) allows multiple independent logical streams to share a common
+underlying transport medium, which amortizes the overhead of establishing new connections with peers in a P2P network.
+In particular, `libp2p` relies on "[stream multiplexing](https://docs.libp2p.io/concepts/stream-multiplexing/)", which
+uses logically distinct "paths" to route requests to the proper handlers. A familiar example of stream multiplexing
+exists in the TCP/IP stack, where unique port numbers are used to distinguish logically independent streams that share a
+common physical transport medium. Gossamer uses [Yamux](#yamux) for stream multiplexing.
+
+## Gossamer Network Protocols
+
+### `libp2p`-Integrated Protocols
+
+#### `ping`
+
+This is a simple liveness check [protocol](https://docs.libp2p.io/concepts/protocols/#ping) that peers can use to
+quickly see if another peer is online - it is
+[included](https://github.com/libp2p/go-libp2p/tree/master/p2p/protocol/ping) with the official Go implementation of
+`libp2p`.
+
+#### `identify`
+
+The [`identify` protocol](https://docs.libp2p.io/concepts/protocols/#identify) allows peers to exchange information
+about each other, most notably their public keys and known network addresses; like [`ping`](#ping), it is
+[included with `go-libp2p`](https://github.com/libp2p/go-libp2p/tree/master/p2p/protocol/identify).
+
+#### Noise
+
+As described above, [Noise](http://noiseprotocol.org/) provides `libp2p` with its
+[key distribution](#identity--key-management) capabilities. The Noise protocol is
+[well documented](http://cryptowiki.net/index.php?title=Noise_Protocol_Framework) and the Go implementation is
```

Comment:
it might be nicer to link to the libp2p spec: https://github.com/libp2p/specs/blob/master/noise/README.md afaik the libp2p version of noise and the "official" version have some slight differences